### PR TITLE
fix(FN-976): fix report download link

### PR DESCRIPTION
**Pull request by toddym42**
## Introduction

Various fixes to get utilisation report downloads working:

* Updated `utilisation-reports/:_id` endpoint in central API to start with a `/`
* Send all report details on `previous-reports` endpoint and do mapping in UI layer (previously only had month name and (now removed) path, so couldn't download report by ID and set correct filename in anchor link `download` attribute)
* changed return type of `api.downloadUtilisationReport` to include entire axios response instead of just the data, so headers can also be used to populate the file `Content-Disposition` and `Content-Type`
* Fixed `validateBankIdForUser` in Portal UI to get user from `req.session` instead of just `req` (which is how we access in the API layer)

### CODE REVIEW

File: portal-api/api-tests/v1/utilisation-report-service/previous-reports.api-test.js
Comment (JoshBinns2000) [inline portal-api/api-tests/v1/utilisation-report-service/previous-reports.api-test.js:34]: I'm slightly concerned about doing it this way as the reports we actually feed into the database use a `Date` object for `dateUploaded` rather than an ISO string. It might be better to use a function in line 88 to convert the dates in `expectedResponse` to their respective ISO strings. Alternatively, we can use a function in `JSON.parse` to re-convert ISO strings to `Date` objects like this https://stackoverflow.com/questions/4511705/how-to-parse-json-to-receive-a-date-object-in-javascript

File: portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.test.js
Comment (JoshBinns2000) [inline portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.test.js:1]: Does this file not need to update `reports` to have `azureFileInfo` rather than `path`?

File: portal/server/controllers/utilisation-report-service/utilisation-report-download/index.js
Comment (JoshBinns2000) [inline portal/server/controllers/utilisation-report-service/utilisation-report-download/index.js:11]: In one of the api tests above, you use `'content-disposition'` and `'content-type'` for the headers instead of the capitalised versions. Which one are you expecting to use?

File: portal/api-tests/utilisation-report-routes.api-test.js
Comment (JoshBinns2000) [inline portal/api-tests/utilisation-report-routes.api-test.js:14]: Linked to the comment below

**Review by JoshBinns2000 — COMMENTED** [review]
Looks mostly fine - just a few points of clarification and suggestions from me

File: portal-api/api-tests/v1/utilisation-report-service/previous-reports.api-test.js
Comment (JoshBinns2000) [inline portal-api/api-tests/v1/utilisation-report-service/previous-reports.api-test.js:103]: Are we sure about removing the check of the return body type? If the date is that big of a pain maybe only check the month, year and azureFileInfo?

File: portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.test.js
Comment (JoshBinns2000) [inline portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.test.js:11]: Should this not also be azureFileInfo?

File: portal-api/api-tests/v1/utilisation-report-service/previous-reports.api-test.js
Comment (toddym42) [reply portal-api/api-tests/v1/utilisation-report-service/previous-reports.api-test.js:103]: I'm pretty sure. Happy to wait for @ChristianMcc to approve instead though if you're uncomfortable with this.

Comment (toddym42) [reply portal-api/api-tests/v1/utilisation-report-service/previous-reports.api-test.js:110]: Found a way to assert on the response without having to manually map all the `ObjectId` (for mongo `_id` values) and `Date` objects! Maybe there is some value in a single happy path test to at least make sure the correct controller is called. Either way I think it would still be interesting to have this discussion.

Comment (JoshBinns2000) [reply portal-api/api-tests/v1/utilisation-report-service/previous-reports.api-test.js:110]: lol I love the hack of stringifying and then posing as JSON dad

File: portal/server/controllers/utilisation-report-service/utilisation-report-download/index.js
Comment (ChristianMcc) [inline portal/server/controllers/utilisation-report-service/utilisation-report-download/index.js:8]: does this function need to be wrapped in try/catch?

File: portal-api/api-tests/v1/utilisation-report-service/previous-reports.api-test.js
Comment (ChristianMcc) [reply portal-api/api-tests/v1/utilisation-report-service/previous-reports.api-test.js:110]: I think I would have been fine with just checking status and treating this as just a green/red on can we get a good response. The stringify/parse hack is nicer though for sure and a nicer form of regression testing so works for me :)

**Review by ChristianMcc — COMMENTED** [review]
LGTM - just one small try/catch check

File: portal/server/controllers/utilisation-report-service/utilisation-report-download/index.js
Comment (ChristianMcc) [inline portal/server/controllers/utilisation-report-service/utilisation-report-download/index.js:5]: CSV*

**Review by ChristianMcc — APPROVED** [review]

**Pull request merged**
